src/Configuration/Dotenv.hs

```diff
@@ -89,13 +91,16 @@ applySetting :: MonadIO m =>
   -> DotEnv m (String, String)
 applySetting kv@(k, v) = do
   Config { .. } <- ask
-  if configOverride
+  if configOverride && allowDuplicates
     then info kv >> setEnv'
     else do
       res <- liftReaderT . liftIO $ lookupEnv k
       case res of
         Nothing -> info kv >> setEnv'
-        Just _  -> return kv
+        Just _  -> onDisallowDuplicates k >>
+                if configOverride
+                  then info kv >> setEnv'
+                  else return kv
```

Comment:
Hey @FranzGB Correct me if I'm wrong but I see your proposition here is to throw an error if the `--no-dups` flag is given and the variable is set twice. Am I write? I suppose we won't get an error if the environment variable is already in the system (thanks to the `configOverride`). However, I found this implementation kind of complicated. I would suggest to reduce the [cyclomatic complexity](https://en.m.wikipedia.org/wiki/Cyclomatic_complexity) by double checking duplicated environment variables one step above (before calling `applySetting` on any variable). WDYT?